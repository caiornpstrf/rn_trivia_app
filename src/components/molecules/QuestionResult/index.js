/**
 * A visual representation of the result of a question.
 * If correct show a positive view, else a negative view.
 *
 * @summary Result of a Question
 * @author Caio Reis <caio.oliveira.reis@gmail.com>
 *
 * Created at     : 2021-04-12 06:36:21
 * Last modified  : 2021-04-16 03:42:21
 */

import React from 'react';
import {bool, string} from 'prop-types';

import {Card, Icon, Title} from '_components/atoms';
import {Colors, StyleHelpers} from '_assets/styles';

import selectStyles from './selectStyles';

const QuestionResult = ({isCorrect, correctLabel, incorrectLabel}) => {
  const styles = selectStyles(isCorrect ? 'correct' : 'incorrect');
  const iconProps = {
    name: isCorrect ? 'check-circle' : 'times-circle',
    color: isCorrect ? Colors.success : Colors.error,
    size: StyleHelpers.normalize(50),
  };

  return (
    <Card style={styles.container}>
      <Icon {...iconProps} />
      <Title size="medium" style={styles.text}>
        {isCorrect ? correctLabel : incorrectLabel}
      </Title>
    </Card>
  );
};

QuestionResult.propTypes = {
  isCorrect: bool,
  correctLabel: string,
  incorrectLabel: string,
};

QuestionResult.defaultProps = {
  isCorrect: false,
  correctLabel: '',
  incorrectLabel: '',
};

export default QuestionResult;
