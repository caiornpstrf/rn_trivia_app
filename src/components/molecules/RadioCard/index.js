/**
 * Combines Atoms Card and Description can be toggled.
 * It is a Molecule in case a expansion is required.
 *
 * @summary Card with a Description that can be toggled.
 * @author Caio Reis <caio.oliveira.reis@gmail.com>
 *
 * Created at     : 2021-04-12 04:16:13
 * Last modified  : 2021-04-12 05:08:05
 */

import React from 'react';
import {StyleSheet, Pressable} from 'react-native';
import {bool, func, string} from 'prop-types';

import {Card, Description} from '_components/atoms';
import {Colors} from '_assets/styles';

const RadioCard = ({description, isSelected, onPress, style}) => {
  return (
    <Pressable onPress={onPress} style={styles.container}>
      <Card style={[style, isSelected && styles.selected]}>
        <Description>{description}</Description>
      </Card>
    </Pressable>
  );
};

RadioCard.propTypes = {
  description: string,
  isSelected: bool,
  onPress: func,
};

RadioCard.defaultProps = {
  description:
    'sdnas jdnasijdn aoisjd oiajsbd oiasbd iahsjbd ioasbdo iasbd iohjasbdi',
  isSelected: false,
  onPress: () => console.log('Pressed RadioCard'),
};

export default RadioCard;

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  selected: {
    borderWidth: 2.5,
    borderColor: Colors.selection,
  },
});
