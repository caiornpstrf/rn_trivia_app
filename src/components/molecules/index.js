export {default as QuestionDetails} from './QuestionDetails';
export {default as QuestionResult} from './QuestionResult';
export {default as RadioCard} from './RadioCard';
