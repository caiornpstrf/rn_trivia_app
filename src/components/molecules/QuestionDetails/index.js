import React from 'react';
import {StyleSheet, Text, View} from 'react-native';
import {oneOf, string} from 'prop-types';

import {Description, DifficultyIndicator, Title} from 'components/atoms';

const QuestionDetails = ({title, question, difficulty}) => {
  return (
    <View style={styles.container}>
      <View style={styles.top}>
        <Title style={styles.title} size="medium">
          {title}
        </Title>
        <DifficultyIndicator currentDifficulty={difficulty} />
      </View>
      <Description>{question}</Description>
    </View>
  );
};

QuestionDetails.propTypes = {
  title: string,
  question: string,
  difficulty: oneOf(['easy', 'medium', 'hard']),
};

QuestionDetails.defaultProps = {
  title: '',
  question: '',
  difficulty: 'easy',
};

export default QuestionDetails;

const styles = StyleSheet.create({
  container: {
    margin: 10,
  },
  top: {
    marginBottom: 10,
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  title: {
    flex: 1,
  },
});
