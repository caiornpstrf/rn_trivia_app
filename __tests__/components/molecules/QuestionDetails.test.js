import React from 'react';
import {render} from '@testing-library/react-native';
import {QuestionDetails} from '_components/molecules';

const baseProps = {
  title: 'Question title',
  question: 'Is this a test question?',
};
const difficulties = {
  easy: 'easy',
  medium: 'medium',
  hard: 'hard',
};

it('renders on easy difficulty correctly', () => {
  const {toJSON} = render(
    <QuestionDetails {...baseProps} difficulty={difficulties.easy} />,
  );
  expect(toJSON()).toMatchSnapshot();
});

it('renders on medium difficulty correctly', () => {
  const {toJSON} = render(
    <QuestionDetails {...baseProps} difficulty={difficulties.medium} />,
  );
  expect(toJSON()).toMatchSnapshot();
});

it('renders on hard difficulty correctly', () => {
  const {toJSON} = render(
    <QuestionDetails {...baseProps} difficulty={difficulties.hard} />,
  );
  expect(toJSON()).toMatchSnapshot();
});
