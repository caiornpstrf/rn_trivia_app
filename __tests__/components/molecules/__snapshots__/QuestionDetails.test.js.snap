// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders on easy difficulty correctly 1`] = `
<View
  style={
    Object {
      "margin": 10,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 10,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#343D58",
            "fontWeight": "bold",
          },
          Object {
            "flex": 1,
          },
          Object {
            "fontSize": 35,
          },
        ]
      }
    >
      Question title
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#D6D8DE",
          "borderRadius": 20,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingHorizontal": 10,
          "paddingVertical": 5,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "marginRight": 10,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#343D58",
                "fontSize": 12,
              },
              undefined,
              Object {
                "fontFamily": "FontAwesome",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#B0B0B0",
                "fontSize": 12,
              },
              undefined,
              Object {
                "fontFamily": "FontAwesome",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#B0B0B0",
                "fontSize": 12,
              },
              undefined,
              Object {
                "fontFamily": "FontAwesome",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
      <Text
        style={
          Object {
            "color": "#343D58",
            "textTransform": "capitalize",
          }
        }
      >
        easy
      </Text>
    </View>
  </View>
  <Text
    style={
      Object {
        "color": "#171717",
        "fontSize": 28,
      }
    }
  >
    Is this a test question?
  </Text>
</View>
`;

exports[`renders on hard difficulty correctly 1`] = `
<View
  style={
    Object {
      "margin": 10,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 10,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#343D58",
            "fontWeight": "bold",
          },
          Object {
            "flex": 1,
          },
          Object {
            "fontSize": 35,
          },
        ]
      }
    >
      Question title
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#D6D8DE",
          "borderRadius": 20,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingHorizontal": 10,
          "paddingVertical": 5,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "marginRight": 10,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#343D58",
                "fontSize": 12,
              },
              undefined,
              Object {
                "fontFamily": "FontAwesome",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#343D58",
                "fontSize": 12,
              },
              undefined,
              Object {
                "fontFamily": "FontAwesome",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#343D58",
                "fontSize": 12,
              },
              undefined,
              Object {
                "fontFamily": "FontAwesome",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
      <Text
        style={
          Object {
            "color": "#343D58",
            "textTransform": "capitalize",
          }
        }
      >
        hard
      </Text>
    </View>
  </View>
  <Text
    style={
      Object {
        "color": "#171717",
        "fontSize": 28,
      }
    }
  >
    Is this a test question?
  </Text>
</View>
`;

exports[`renders on medium difficulty correctly 1`] = `
<View
  style={
    Object {
      "margin": 10,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 10,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#343D58",
            "fontWeight": "bold",
          },
          Object {
            "flex": 1,
          },
          Object {
            "fontSize": 35,
          },
        ]
      }
    >
      Question title
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#D6D8DE",
          "borderRadius": 20,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingHorizontal": 10,
          "paddingVertical": 5,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "marginRight": 10,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#343D58",
                "fontSize": 12,
              },
              undefined,
              Object {
                "fontFamily": "FontAwesome",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#343D58",
                "fontSize": 12,
              },
              undefined,
              Object {
                "fontFamily": "FontAwesome",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#B0B0B0",
                "fontSize": 12,
              },
              undefined,
              Object {
                "fontFamily": "FontAwesome",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
      <Text
        style={
          Object {
            "color": "#343D58",
            "textTransform": "capitalize",
          }
        }
      >
        medium
      </Text>
    </View>
  </View>
  <Text
    style={
      Object {
        "color": "#171717",
        "fontSize": 28,
      }
    }
  >
    Is this a test question?
  </Text>
</View>
`;
